pkg/controller/certificates/sync.go

```diff
@@ -186,6 +186,9 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (reque
 		return c.issue(ctx, i, crtCopy)
 	}
 
+	// update certificate expiry metric
+	defer c.metrics.UpdateCertificateExpiry(crt, c.secretLister)
```

Comment:
I think this should be moved up to around L164 (i.e. `	crtCopy.Status.NotAfter = &metaNotAfter`) - else there may be some cases where we don't update this field whilst a certificate is being issued/renewed.